drivers/kscan/Kconfig.xec

```diff
@@ -45,3 +46,46 @@ config KSCAN_XEC_POLL_PERIOD
 	  Defines the poll period in msecs between between matrix scans.
 
 endif # KSCAN_XEC
+
+menuconfig KSCAN_XEC_V2
+	bool "XEC Microchip KSCAN V2 driver"
+	depends on SOC_SERIES_MEC172X
+	select MULTITHREADING
+	help
+	  Enable the Microchip XEC MEC172x Kscan IO driver.
+
+if KSCAN_XEC_V2
+
+config KSCAN_XEC_COLUMN_SIZE
+	int "KSCAN_XEC_COLUMN_SIZE"
+	default 16
+	help
+	  Adjust the value to your keyboard columns. The maximum
+	  column size for the Microchip XEC family is 18 (from 0 to 17).
+
+config KSCAN_XEC_ROW_SIZE
+	int "KSCAN_XEC_ROW_SIZE"
+	default 8
+	help
+	  Adjust the value to your keyboard rows. The maximum
+	  column size for the Microchip XEC family is 8 (from 0 to 7).
+
+config KSCAN_XEC_DEBOUNCE_DOWN
+	int "KSCAN_XEC_DEBOUNCE_DOWN"
+	default 10
+	help
+	  Determines the time in msecs for debouncing a key press.
+
+config KSCAN_XEC_DEBOUNCE_UP
+	int "KSCAN_XEC_DEBOUNCE_UP"
+	default 20
+	help
+	  Determines the time in msecs for debouncing a key release.
+
+config KSCAN_XEC_POLL_PERIOD
+	int "KSCAN_XEC_POLL_PERIOD"
+	default 5
+	help
+	  Defines the poll period in msecs between between matrix scans.
```

Comment:
can't these be DT properties?

Reply:
These properties are dependent on the keypad connected to the platform. The hardware (kscan) needs to be configured based on that. Hence these properties are application dependent and hence in the Kconfig.

Reply:
DT is the language used to describe hardware, so I still don't understand why DT can't be used for that.

Reply:
keypad is the external device connected to the platform, the kscan driver uses the Soc keyscan hardware to access the external keypad. Hence in DT we are describing keyscan hardware, but keypad details is unknown and depends on application usage and hence is in the kconfig. 

This is similar to [Kconfig.it8xxx2](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/kscan/Kconfig.it8xxx2)

Reply:
Fair enough, thanks for the explanation. I guess it's ok then.